src/nnet3/nnet-utils.cc

```diff
@@ -654,6 +654,32 @@ void ReadEditConfig(std::istream &edit_config_is, Nnet *nnet) {
         }
       }
       KALDI_LOG << "Set learning rates for " << num_learning_rates_set << " nodes.";
+    } else if (directive == "set-learning-rate-factor") {
+      std::string name_pattern = "*";
+      // name_pattern defaults to '*' if none is given.
+      config_line.GetValue("name", &name_pattern);
+      BaseFloat learning_rate_factor = -1;
+      if (!config_line.GetValue("learning-rate-factor", &learning_rate_factor)) {
+        KALDI_ERR << "In edits-config, expected learning-rate-factor to be set in line: "
+                  << config_line.WholeLine();
+      }
+      // Note: the learning_rate_factor_  defined in the component
+      // sets to the value you provided, so if you call SetUnderlyingLearningRate(),
+      // the actual learning rate (learning_rate_) is set to the value you provided
+      // times learning_rate.
+      UpdatableComponent *component = NULL;
+      int32 num_learning_rate_factors_set = 0;
+      for (int32 c = 0; c < nnet->NumComponents(); c++) {
+        if (NameMatchesPattern(nnet->GetComponentName(c).c_str(),
+            name_pattern.c_str()) &&
+            (component =
+            dynamic_cast<UpdatableComponent*>(nnet->GetComponent(c)))) {
+          component->SetLearningRateFactor(learning_rate_factor);
+          num_learning_rate_factors_set++;
+        }
+      }
+      KALDI_LOG << "Set learning rate factors for " << num_learning_rate_factors_set
+                << " nodes.";
```

Comment:
change 'nodes' to 'components' since the learning rate factor applies to components, not nodes.